libs/blocks/gnav/gnav-breadcrumbs.js

```diff
@@ -0,0 +1,156 @@
+import { createTag, getMetadata } from '../../utils/utils.js';
+
+const BREADCRUMBS_HIDE_LAST = 'breadcrumbs-hide-last';
+
+function getParent(path) {
+  if (!path || path.length === 0 || path === '/') {
+    return null;
+  }
+  const elements = path.split('/').filter((n) => n);
+  elements.pop();
+  if (elements.length === 0) {
+    return '/';
+  }
+  return `/${elements.join('/')}`;
+}
+
+function getItemTitle(doc, defaultTitle) {
+  const title = getMetadata('breadcrumbs-title', doc);
+  if (title) return title;
+  const titleElt = doc.getElementsByTagName('title');
+  if (!titleElt || !titleElt[0]) return defaultTitle;
+  return titleElt[0].innerHTML;
+}
+
+async function getItem(path) {
+  if (path === null) return null;
+  const defaultTitle = path.split('/').pop();
+  const defaultItem = {
+    path,
+    title: defaultTitle,
+  };
+  const res = await fetch(path);
+  if (!res.ok) return defaultItem;
+  const html = await res.text();
+  if (!html) return defaultItem;
+  const doc = new DOMParser().parseFromString(html, 'text/html');
+  if (!doc) return defaultItem;
+  const hideInNav = getMetadata('breadcrumbs-hide-page', doc) === 'on';
+  const hideLastPage = getMetadata(BREADCRUMBS_HIDE_LAST, doc) === 'on';
+  return {
+    path,
+    title: getItemTitle(doc, defaultTitle),
+    hideInNav,
+    hideLastPage,
+  };
+}
+
+async function getItems(pagePath) {
+  const items = [];
+  let path = pagePath;
+  while (path !== null) {
+    // eslint-disable-next-line no-await-in-loop
+    const item = await getItem(path);
+    if (item) items.push(item);
+    path = getParent(path);
+  }
+  items.reverse();
+  return items;
+}
+
+async function getBreadcrumbsFromUrl(pagePath) {
+  const items = await getItems(pagePath);
+  if (!items || items.length === 0) return null;
+  const ul = createTag('ul');
+  const last = items.length - 1;
+  items.forEach((item, idx) => {
+    if (item.hideInNav) return;
+    const li = createTag('li');
+    if (idx === last) {
+      if (!item.hideLastPage) {
+        li.textContent = `${item.title}`;
+        li.setAttribute('aria-current', 'page');
+      }
+    } else {
+      const a = createTag('a', { href: item.path });
+      a.textContent = `${item.title}`;
+      li.append(a);
+    }
+    ul.append(li);
```

Comment:
I noticed a trailing slash in case `Breadcrumbs Hide Last` is set to `on` which I don't think is correct. An additional check needs to be added here.

Reply:
@honstar IMHO this is the correct behaviour. E.g.:
1) the requested page is `/foo/bar`
2a) the last breadcrumb item is not hidden
-> the breadcrumb should render as: `/ foo / bar `
where `foo` is a link, `bar` is the page title (not linked)
2b) the last breadcrumb item is hidden
-> the breadcrumb should render as: `/ foo /`
where `foo` is a link

Reply:
Hmm, you have a point there @jckautzmann. @chrischrischris @auniverseaway , wdyt?

Reply:
Let's keep it as-is, we can always change it if needed.